frontend/lib/DataModels/User/user.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:adpv_frontend/DataModels/User/auth_token_response.dart';
+import 'package:adpv_frontend/DataModels/User/user_role.dart';
+import 'package:jwt_decode/jwt_decode.dart';
+
+class User {
+  late AuthTokenResponse tokenResponse;
+  late String email;
+  List<UserRole> userRole = [];
+
+  User(this.tokenResponse) {
+    final Map<String, dynamic> parsed = Jwt.parseJwt(tokenResponse.accessToken);
+    email = parsed['sub'];
+    for (var element in (parsed['roles'] as List)) {
+      userRole.add(UserRoleExtension.userRole(element));
+    }
```

Comment:
Można od razu stworzyć listę ról, zamiast dodawać do niej kolejne elementy. Wtedy `List<UserRole> userRole` może mieć przedrostek `late`
```suggestion
userRole = (parsed['roles'] as List<String>)
        .map<UserRole>(UserRoleExtension.userRole)
        .toList();
```


Wg tutaj chyba nie powinniśmy używać `UserRoleExtension`. Rolą extensionów jest rozszerzenie jakiejś klasy o dodatkowe metody. W tym przypadku jeśli chciałbyś użyć extension'a, to trzeba wywołać `UserRole.userRole` zamiast `UserRoleExtension.userRole`, bo my rozszerzamy klasę `UserRole` o metodę `userRole`.  W [dokumentacji Flutter'a](https://dart.dev/guides/language/extension-methods) są przykłady, które pewnie lepiej to opisać niż ja

Reply:
no niestety tutaj nie działa sugestia, zostaje for